src/main/java/duke/chatbot/ChatBot.java

```diff
@@ -0,0 +1,149 @@
+package duke.chatbot;
+
+import duke.taskmanager.TaskManager;
+import duke.taskmanager.task.*;
+import duke.taskmanager.exceptions.*;
+import java.util.InputMismatchException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+public class ChatBot {
+    private final String name;
+    private boolean isRunning;
+    private final TaskManager taskManager;
+
+    /**
+     * Constructor for a chatbot that can be intialized or terminated.
+     * Primarily used for processing commands to update its own task manager.
+     * Default state of the chat bot is runningState = false
+     *
+     * @param name string of the name of the chatbot
+     */
+    public ChatBot(String name) {
+        this.name =  name;
+        this.isRunning = false;
+        this.taskManager = new TaskManager();
+    }
+
+    /**
+     * Initializes the chatbot by setting its runningState to true and responds
+     * with a greeting message. The task manager is also initialized by loading
+     * any pre-existing data.
+     */
+    public void initialize() {
+        this.isRunning = true;
+        System.out.println(wrapMessage("Greetings, " + this.name + " at your service.\n"
+                + "How may I help you today?\n"));
+        taskManager.load();
+    }
+
+    /**
+     * Terminates the chatbot by setting its runningState to false and responds
+     * with a goodbye message.
+     */
+    public void terminate() {
+        this.isRunning = false;
+        System.out.println(wrapMessage("Goodbye! It was nice seeing you.\n"));
+    }
+
+    /**
+     * Returns the runningState of the chatbot
+     *
+     * @return running state of the chatbot
+     */
+    public boolean isRunning() {
+        return this.isRunning;
+    }
+
+    /**
+     * Processes commands from an input by the user and calls the appropriate command
+     * for the task manager. Handles exceptions for invalid commands. Calls the
+     * save command of the task manager after every successful command processed.
+     *
+     * @param input string of the input provided by the user
+     */
+    public void processCommand(String input) {
+        Scanner inputScanner = new Scanner(input);
+        try {
+            String command = inputScanner.next();
+            if (!(inputScanner.hasNext())) {
+                switch (command) {
+                case "bye":
+                    this.isRunning = false;
+                    break;
+                case "list":
+                    System.out.println(wrapMessage(taskManager.list()));
```

Comment:
Not sure I like the wrapMessage() here. It might be a good idea to abstract all the System.out.println() methods with an UI class.